tests_integration/nativeapp/test_debug_mode.py

```diff
@@ -57,24 +57,39 @@ def set_yml_application_debug(snowflake_yml: Path, debug: Optional[bool]):
     """
     pdf = yaml.load(snowflake_yml.read_text(), yaml.BaseLoader)
 
-    if "native_app" not in pdf:
-        pdf["native_app"] = dict()
-
-    if "application" not in pdf["native_app"]:
-        pdf["native_app"]["application"] = dict()
-
-    if debug is None and "debug" in pdf["native_app"]["application"]:
-        del pdf["native_app"]["application"]["debug"]
-    elif debug is not None:
-        pdf["native_app"]["application"]["debug"] = debug
+    if pdf["definition_version"] == "2":
```

Comment:
Nit: could `defaultdict` help here?

Reply:
Maybe I don't understand the suggestion, but I'm not sure it will really simplify the code. Can you explain what you mean?